LocaleEmulator/Hooks/Kernel32Hook.cpp

```diff
@@ -123,14 +123,107 @@ NTSTATUS LeGlobalData::HackUserDefaultLCID2(PVOID Kernel32)
     return STATUS_SUCCESS;
 }
 
+void* GetFirstCallTarget(void* start_offset, DWORD parse_range, void **next) {
+    void* res = nullptr;
+    WalkOpCodeT(start_offset, parse_range,
+        WalkOpCodeM(Buffer, OpLength, Ret)
+    {
+        if (Buffer[0] != CALL) {
+            return STATUS_NOT_FOUND;
+        }
+
+        res = GetCallDestination(Buffer);
+        *next = &Buffer[OpLength];
+        return STATUS_SUCCESS;
+    }
+    );
+    return res;
+}
+
+void* GetKthCallTarget(void* start_offset, DWORD parse_range_each, int K) {
+    if (start_offset == nullptr)
+        return nullptr;
+    void *next, *res;
+    while (K >= 1) {
+        res = GetFirstCallTarget(start_offset, parse_range_each, &next);
+        if (res == nullptr)
+            return nullptr;
+        start_offset = next;
+        --K;
+    }
+    return res;
+}
+
+typedef DWORD(__stdcall* pSetupAnsiOemCodeHashNodes)();
+
+NTSTATUS LeSetupAnsiOemCodeHashNodes() {
```

Comment:
Could you add two `WriteLog(L"xxx")` commands at the beginning and end of this method? It'll make future debugging easier.

Reply:
Resolved.